doc/bhyve.md

```diff
@@ -0,0 +1,204 @@
+<li>Use diskN to specify the slot into which the disk will be placed. A plain <i>disk</i> tag will be put in the lowest available slot.</li>
+<li>Available firmware files can be found in <i>/usr/share/bhyve/firmware/</i>;</li>
+<li>The ISO file needs passing through to the zone via a lofs mount, see the example below;</li>
+<li>Setting vnc to <i>on</i> is the same as setting it to <i>unix=/tmp/vm.vnc</i>.</li>
+<li>You can connect to the virtual machine console from the global zone with <i>zlogin -C zonename</i>;</li>
+</ol>
+
+### Example zone
+
+The following example zone is shown twice, once in info format and once in
+export (showing the necessary commands for creation). Note that the example
+shows setting the `allowed-address` attribute for the network interface -
+this does not configure the address within the virtual machine but rather
+prevents the use of any other address (L3 protection).
+
+```
+hipster# zonecfg -z bhyve info
+zonename: bhyve
+zonepath: /data/zone/bhyve
+brand: bhyve
+autoboot: false
+bootargs:
+pool:
+limitpriv:
+scheduling-class:
+ip-type: exclusive
+hostid:
+fs-allowed:
+fs:
+        dir: /tank/iso/FreeBSD-11.1-RELEASE-amd64-bootonly.iso
+        special: /tank/iso/FreeBSD-11.1-RELEASE-amd64-bootonly.iso
+        raw not specified
+        type: lofs
+        options: [ro,nodevices]
+net:
+        address not specified
+        allowed-address: 10.0.0.112/24
+        defrouter not specified
+        physical: bhyve0
+device:
+        match: /dev/zvol/rdsk/tank/hdd/bhyve0
+device:
+        match: /dev/zvol/rdsk/tank/hdd/bhyve1
+device:
+        match: /dev/zvol/rdsk/tank/hdd/bhyve2
+attr:
+        name: vcpus
+        type: string
+        value: cpus=16,sockets=2,cores=4,threads=2
+attr:
+        name: ram
+        type: string
+        value: 4G
+attr:
+        name: bootrom
+        type: string
+        value: BHYVE_DEBUG
+attr:
+        name: console
+        type: string
+        value: socket,/tmp/vm.com1
+attr:
+        name: hostbridge
+        type: string
+        value: amd
+attr:
+        name: bootdisk
+        type: string
+        value: tank/hdd/bhyve0
+attr:
+        name: disk
+        type: string
+        value: tank/hdd/bhyve1
+attr:
+        name: disk1
+        type: string
+        value: tank/hdd/bhyve2,serial=1234
+attr:
+        name: cdrom
+        type: string
+        value: /tank/iso/FreeBSD-11.1-RELEASE-amd64-bootonly.iso
+attr:
+        name: vnc
+        type: string
+        value: unix=/tmp/vm.vnc,wait
+```
+
+```
+hipster# zonecfg -z bhyve export
+create -b
+set zonepath=/data/zone/bhyve
+set brand=bhyve
+set autoboot=false
+set ip-type=exclusive
+add fs
+set dir=/tank/iso/debian-9.4.0-amd64-netinst.iso
+set special=/tank/iso/debian-9.4.0-amd64-netinst.iso
+set type=lofs
+add options ro
+add options nodevices
+end
+add net
+set allowed-address=10.0.0.112/24
```

Comment:
This is likely where difference comes. Have you rechecked allowed-address option and net-related zonecfg output?

Reply:
Yes these are valid attributes I use on a regular basis.